Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +22,72 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return totalAmphoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(current_ < 0){
+        return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double totalHoursRemaining, msRemaining;
+        int h, m, s, ms;
+
+        if(isCharging())
+            totalHoursRemaining = totalAmpHoursUsed() / current_ * -1;
+        else
+            totalHoursRemaining = (BATTERY_AMP_HOUR_CAPACITY - totalAmpHoursUsed()) / current_;
+
+        while(totalHoursRemaining > 24){
+            totalHoursRemaining -= 24;
+        }
+        //Converting totalHoursRemaining into h, m, s, ms
```

Comment:
Just a reminder to remove unnecessary comments, want to make code that is self-documenting.